From 1afa35bd98c71e48e87ef093dc8a10c70ad65c3f Mon Sep 17 00:00:00 2001
From: Matthew Gerlach <matthew.gerlach@linux.intel.com>
Date: Wed, 2 Jun 2021 11:55:07 -0700
Subject: [PATCH 89/93] fpga: dfl: add debug to failed partial reconfiguration

Add dev_err call printing number of bytes pushed out of total
attempted after a failed partial reconfigation.

Signed-off-by: Matthew Gerlach <matthew.gerlach@linux.intel.com>
---
 drivers/fpga/dfl-fme-mgr.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/fpga/dfl-fme-mgr.c b/drivers/fpga/dfl-fme-mgr.c
index 4a12a988ab38..2969c6f59d31 100644
--- a/drivers/fpga/dfl-fme-mgr.c
+++ b/drivers/fpga/dfl-fme-mgr.c
@@ -180,6 +180,7 @@ static int fme_mgr_write(struct fpga_manager *mgr,
 	void __iomem *fme_pr = priv->ioaddr;
 	u64 pr_ctrl, pr_status, pr_data;
 	int delay = 0, pr_credit;
+	size_t full_cnt = count;
 	size_t chunk_size;
 
 	dev_dbg(dev, "start request\n");
@@ -203,6 +204,8 @@ static int fme_mgr_write(struct fpga_manager *mgr,
 		while (pr_credit <= 1) {
 			if (delay++ > PR_WAIT_TIMEOUT) {
 				dev_err(dev, "PR_CREDIT timeout\n");
+				dev_err(dev, "wrote %zu bytes of %zu total\n",
+					full_cnt - count, full_cnt);
 				return -ETIMEDOUT;
 			}
 			udelay(1);
-- 
2.39.0

